src/index.js

```diff
@@ -38,35 +54,38 @@ function loadConfig(configPath) {
       'lint-staged.config.js'
     ]
   })
+  const resultPromise = configPath ? explorer.load(resolveConfig(configPath)) : explorer.search()
+
+  return resultPromise.then(result => {
```

Comment:
Let's use `async/await` here to make the code more easy to understand. Also this is missing error handling.